MULTI_UNDERLAY_PROTOCOL.md

```diff
@@ -0,0 +1,254 @@
+# Multi-Underlay Protocol Documentation
+
+## Overview
+
+The multi-underlay protocol is a channel-layer feature that allows a single logical channel to maintain multiple concurrent underlying transport connections (underlays). This provides resilience, load distribution, and the ability to use different transport types simultaneously.
+
+## Core Concepts
+
+### Multi-Channel
+A `MultiChannel` represents a single logical communication channel that can contain multiple underlying transport connections. It implements the standard `Channel` interface while internally managing multiple `Underlay` instances.
+
+### Underlay
+An `Underlay` represents a single underlying transport connection (TCP, WebSocket, UDP, etc.). Multiple underlays can be grouped together under a single multi-channel using a shared group secret.
+
+### Group Secret
+A cryptographic token that identifies which underlays belong to the same logical channel. All underlays sharing the same group secret are considered part of the same multi-channel group.
```

Comment:
I took a quick look to see what cryptographic algorithms are used to generate these symmetric secrets. I could not find an implementation. It seems like it is up to the caller to decide the value.

Reply:
Mmm, I should fix that. It's not cryptographic, it's just a uuid used as a shared secret.